resources/aws/elb.go

```diff
@@ -0,0 +1,97 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elb"
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+// ELB is an Elastic Load Balancer
+type ELB struct {
+	Name             string
+	AZ               string
+	SecurityGroup    string
+	Tags             []string
+	Client           *elb.ELB
+	LoadBalancerPort int
+	InstancePort     int
+}
+
+func (lb *ELB) CreateIfNotExists() (bool, error) {
+	if lb.Client == nil {
+		return false, microerror.MaskAny(clientNotInitializedError)
+	}
+
+	if err := lb.CreateOrFail(); err != nil {
+		if err.Error() == awsclient.ELBAlreadyExists {
+			return false, nil
+		}
+
+		return false, microerror.MaskAny(err)
+	}
+
+	return true, nil
+}
+
+func (lb *ELB) CreateOrFail() error {
+	if lb.Client == nil {
+		return microerror.MaskAny(clientNotInitializedError)
+	}
+
+	if _, err := lb.Client.CreateLoadBalancer(&elb.CreateLoadBalancerInput{
+		LoadBalancerName: aws.String(lb.Name),
+		Listeners: []*elb.Listener{
+			{
+				InstancePort:     aws.Int64(int64(lb.InstancePort)),
+				LoadBalancerPort: aws.Int64(int64(lb.LoadBalancerPort)),
+				// TCP because we want to do SSL offloading, not termination
```

Comment:
According to [AWS' terminology](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-listener-config.html), this is offloading:

> If you don't want the load balancer to handle the SSL termination (known as SSL offloading), you can use TCP for both the front-end and back-end connections, and deploy certificates on the registered instances handling requests.